test/state_test.js

```diff
@@ -161,6 +171,23 @@ describe('State', () => {
     assume(await db.listSpotRequests()).has.length(0);
     assume(await db.listInstances()).has.length(1);
   });
+   
+  it('should be able to do a spot request upsert', async () => {
```

Comment:
so it looks like this unit test moved.  This is not a huge deal in any way, but it does make the review a little more complicated because i want to double check that the code that was re-added is the same.  It also makes future merges a lot easier to work on because there's less chance of a merge conflict from an unintentional change.

A couple tricks for avoiding this is `git revert -p`, which will let you selectively revert a commit.  This is basically git generating the reverse of your diff (`patch -R`), then going through each hunk and asking "want to revert this bit"?  It's a really cool feature, and `-p` is supported in lots of other places too!

Reply:
thanks for the info @jhford , will play around with this and fix today

Reply:
@jhford, with the help of @walac I managed to rebase and do some squashing. I think I even managed to leave this unit test in the same place as it was before, let me know what you think